File: Python/SelectionSort.py
def selectionSort(l):
    for start in range(len(l)):

        #assign start value as minimum value
        minpos=start
        
        for i in range(start,len(l)):
            if(l[i]<l[minpos]):
                minpos=i
            
        (l[start],l[minpos]) = (l[minpos],l[start])


if __name__ == "__main__":
    print('Enter sequence(separated by spaces): ')
    arr=[int(x) for x in input().split()]
    selectionSort(arr)
    print(arr)

File: Ruby/selection_sort.rb
def selection_sort(arr)
  n = arr.length
  (0...n).each do |i|
    ((i + 1)...n).each do |j|
      arr[j], arr[i] = arr[i], arr[j] if arr[j] < arr[i]
    end
  end
  arr
end